File: 19941.py
n, k = map(int, input().split())
where = list(input())
count = 0

for i in range(n):
    if where[i] == 'P':
        for j in range(i-k, i+k+1):
            if 0 <= j < n:
                if where[j] == 'H':
                    where[j] = '0'
                # print("where: ", j, where)
                    count += 1
                    break
print(count)

File: 19941/19941.js
//문제: 19941
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let input = fs.readFileSync(filePath).toString().trim().split("\n");

solution(input);
function solution(input) {
  const [a, k] = input[0].split(" ").map(Number);
  let answer = 0;
  const arr = input[1].split("");
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] === "H") {
      const arr2 =
        i >= k ? arr.slice(i - k, i + k + 1) : arr.slice(0, i + k + 1);
      if (arr2.includes("P")) {
        answer++;
        i >= k
          ? (arr[i + arr2.indexOf("P") - k] = ".")
          : (arr[arr2.indexOf("P") + i] = ".");
      }
    }
  }
  console.log(answer);
}
